x-pack/test/ingest_manager_api_integration/apis/agent_policy/agent_policy_with_agents_setup.ts

```diff
@@ -0,0 +1,170 @@
+      .set('kbn-xsrf', 'xx')
+      .set('Authorization', `ApiKey ${agentAccessAPIKey}`)
+      .send({
+        events: [],
+      })
+      .expect(200);
+
+    expect(checkinApiResponse.actions).length(1);
+
+    return checkinApiResponse.actions[0];
+  }
+
+  // Test all the side effect that should occurs when we create|update an agent policy
+  describe('ingest_manager_agent_policies_with_agents_setup', () => {
+    skipIfNoDockerRegistry(providerContext);
+
+    before(async () => {
+      await esArchiver.loadIfNeeded('fleet/agents');
+    });
+    after(async () => {
+      await esArchiver.unload('fleet/agents');
+    });
+
+    setupIngest(providerContext);
+
+    describe('POST /api/fleet/agent_policies', () => {
+      it('should work and create an enrollment key for the policy', async () => {
+        const name = `test-${Date.now()}`;
+
+        const res = await supertest
+          .post(`/api/fleet/agent_policies`)
+          .set('kbn-xsrf', 'xxxx')
+          .send({
+            name,
+            namespace: 'default',
+          })
+          .expect(200);
+
+        const policyId = res.body.item.id;
+        const enrollmentKey = await getEnrollmentKeyForPolicyId(policyId);
+        expect(enrollmentKey).not.empty();
+      });
+
+      it('should create an agent action `POLICY_CHANGE` for the policy', async () => {
+        const name = `test-${Date.now()}`;
+
+        const res = await supertest
+          .post(`/api/fleet/agent_policies?sys_monitoring=true`)
+          .set('kbn-xsrf', 'xxxx')
+          .send({
+            name,
+            namespace: 'default',
+          })
+          .expect(200);
+
+        const policyId = res.body.item.id;
+        const enrollmentKey = await getEnrollmentKeyForPolicyId(policyId);
+        expect(enrollmentKey).not.empty();
+
+        const action = await getAgentActionsForEnrollmentKey(enrollmentKey.api_key);
+
+        expect(action.type).to.be('POLICY_CHANGE');
+        const agentPolicy = action.data.policy;
+        expect(agentPolicy.id).to.be(policyId);
+        // should have system inputes
+        expect(agentPolicy.inputs).length(2);
+        // should have default output
+        expect(agentPolicy.outputs.default).not.empty();
+      });
+    });
+
+    describe('POST /api/fleet/agent_policies/copy', () => {
+      const TEST_POLICY_ID = `policy1`;
+      it('should work and create an enrollment key for the new policy', async () => {
+        const name = `test-${Date.now()}`;
+
+        const res = await supertest
+          .post(`/api/fleet/agent_policies/${TEST_POLICY_ID}/copy`)
+          .set('kbn-xsrf', 'xxxx')
+          .send({
+            name,
+            description: 'Test',
+          })
+          .expect(200);
+
+        const policyId = res.body.item.id;
+        const enrollmentKey = await getEnrollmentKeyForPolicyId(policyId);
+        expect(enrollmentKey).not.empty();
+      });
```

Comment:
similar question here - why do we have two tests for this, one without `POLICY_CHANGE` assertion?

Reply:
I tried to have two tests, to test separatly the action creation and the enrollment key creation, but as we need the enrollment key to get the action it probably make no sense